tox.ini

```diff
@@ -0,0 +1,55 @@
+[tox]
+isolated_build = True
+envlist = py38, py39, py310
+
+[testenv]
+deps = -e .
```

Comment:
I'm still confused about the  relationship between this and pyproject.toml's `[dev]` optional dependency -- maybe your new comments will clarify. But are both necessary?

Reply:
Answering myself: no, they're not. We'll use `extras = dev` to make sure tox uses the correct optional dependencies.